scripts/test_sensitivity.py

```diff
@@ -0,0 +1,171 @@
+# Extract spectral parameters
+dFdE, crab_par = crab_hegra(E)
+dFdEd0, proton_par = proton_bess(E)
+
+# Rates and weights
+rate_g = mc.rate(mc_par_g['emin'], mc_par_g['emax'], mc_par_g['sp_idx'], \
+              mc_par_g['cone'], mc_par_g['area_sim'], crab_par['f0'], crab_par['e0'])
+
+rate_p = mc.rate(mc_par_p['emin'], mc_par_p['emax'], mc_par_p['sp_idx'], \
+              mc_par_p['cone'], mc_par_p['area_sim'], proton_par['f0'], proton_par['e0'])
+
+
+w_g = mc.weight(mc_par_g['emin'], mc_par_g['emax'], mc_par_g['sp_idx'],
+                crab_par['alpha'], rate_g, mc_par_g['sim_ev'], crab_par['e0'])
+
+w_p = mc.weight(mc_par_p['emin'], mc_par_p['emax'], mc_par_p['sp_idx'],
+                proton_par['alpha'], rate_p, mc_par_p['sim_ev'], proton_par['e0'])
+
+
+e_reco_gw = ((e_reco_g / crab_par['e0'])**(crab_par['alpha'] - mc_par_g['sp_idx'])) \
+            * w_g
+e_reco_pw = ((e_reco_p / proton_par['e0'])**(proton_par['alpha'] - mc_par_g['sp_idx'])) \
+        * w_p
+
+# Arrays to contain the number of gammas and hadrons for different cuts
+
+#Before cuts:
+pre_gamma = np.ndarray(shape=(eb, gb, tb))
+pre_hadron = np.ndarray(shape=(eb, gb, tb))
+
+#After cuts:
+final_gamma = np.ndarray(shape=(eb, gb, tb))
+final_hadrons = np.ndarray(shape=(eb, gb, tb))
+
+
+#Weight events and count nº of events per bin:
+
+for i in range(0,eb):  # binning in energy
+    for j in range(0,gb):  # cut in gammaness
+        for k in range(0,tb):  # cut in theta2
+            eg_w_sum = np.sum(e_reco_gw[(e_reco_g < E[i+1].to_value()) & (e_reco_g > E[i].to_value()) \
+                                       & (gammaness_g > g[j]) & (theta2_g < t[k])])
+
+            ep_w_sum = np.sum(e_reco_pw[(e_reco_p < E[i+1].to_value()) & (e_reco_p > E[i].to_value()) \
+                                        & (gammaness_p > g[j]) & (theta2_p < t[k])])
+
+            pre_gamma[i][j][k] = e_reco_g[(e_reco_g < E[i+1].to_value()) & (e_reco_g > E[i].to_value()) \
+                                          & (gammaness_g > g[j]) & (theta2_g < t[k])].shape[0]
+
+            pre_hadron[i][j][k] = e_reco_p[(e_reco_p < E[i+1].to_value()) & (e_reco_p > E[i].to_value()) \
+                                          & (gammaness_p > g[j]) & (theta2_p < t[k])].shape[0]
+
+            final_gamma[i][j][k] = eg_w_sum * obstime.to_value()
+            final_hadrons[i][j][k] = ep_w_sum * obstime.to_value()
+
+sens = sensitivity.calculate_sensitivity(final_gamma, final_hadrons, 1/noff)
+
+#Avoid bins which are empty or have too few events:
+
+min_num_events = 10 #Minimum number of gamma and proton events in a bin to be taken into
+#account for minimization
+
+for i in range(0, eb):
+    for j in range(0, gb):
+        for k in range(0, tb):
+            if np.isnan(sens[i][j][k]) or np.isinf(sens[i][j][k]) or sens[i][j][k]==0:
+                sens[i][j][k] = 1e100
+            if pre_gamma[i][j][k] < min_num_events or pre_hadron[i][j][k] < min_num_events:
+                sens[i][j][k] = 1e100
```

Comment:
this condition should be applied to `final_gamma[i][j][k]` and `final_hadrons[i][j][k]`

Reply:
you can also add the condition that `final_gamma[i][j][k] > final_hadrons[i][j][k] * 0.05` (the excesses need to be > 5% of the background)